datumaro/plugins/voc_format/format.py

```diff
@@ -124,181 +129,278 @@ class VocPath:
+if sys.version_info.minor < 9:
+    _voc_label_map_base = OrderedDict
+else:
+    _voc_label_map_base = OrderedDict[str, VocLabelInfo]
 
 
-def parse_label_map(path: str) -> LabelMapConfig:
+class VocLabelMap(_voc_label_map_base):
+    """
+    Provides VOC-specific info about labels.
+    A mapping of type: str -> VocLabelInfo
     """
-    Parses a label map file in the format:
-    'name : color (r, g, b) : parts (hand, feet, ...) : actions (siting, standing, ...)'
 
-    Parameters:
-        path: File path
+    DEFAULT_BACKGROUND_LABEL = "background"
+    DEFAULT_BACKGROUND_COLOR: RgbColor = (0, 0, 0)
 
-    Returns:
-        A dictionary: label -> (color, parts, actions)
-    """
+    def __init__(self, *args, **kwargs) -> None:
+        d = OrderedDict(*args, **kwargs)  # backward compatibility
+
+        super().__init__(
+            (k, v if isinstance(v, VocLabelInfo) else VocLabelInfo(*v)) for k, v in d.items()
+        )
+
+    def is_label(self, label: str) -> bool:
+        return self.get(label) is not None
+
+    def is_part(self, part: str) -> bool:
+        for label_desc in self.values():
+            if part in label_desc.parts:
+                return True
+        return False
+
+    def is_action(self, label: str, action: str) -> bool:
+        return action in self.get_actions(label)
 
-    label_map = OrderedDict()
-    with open(path, "r", encoding="utf-8") as f:
-        for line in f:
-            # skip empty and commented lines
-            line = line.strip()
-            if not line or line[0] == "#":
-                continue
-
-            # name : color : parts : actions
-            label_desc = line.strip().split(":")
-            if len(label_desc) != 4:
-                raise InvalidAnnotationError(
-                    f"Label description has wrong number of fields '{len(label_desc)}'. "
-                    "Expected 4 ':'-separated fields."
-                )
-            name = label_desc[0]
-
-            if name in label_map:
-                raise InvalidAnnotationError(f"Label '{name}' is already defined in the label map")
-
-            if 1 < len(label_desc) and label_desc[1]:
-                color = label_desc[1].split(",")
-                if len(color) != 3:
+    def get_actions(self, label: str) -> Set[str]:
+        label_desc = self.get(label)
+        if not label_desc:
+            return set()
+        return label_desc.actions
+
+    def has_colors(self) -> bool:
+        return any(v.color is not None for v in self.values())
+
+    def find_background_label(
+        self,
+        *,
+        name: str = DEFAULT_BACKGROUND_LABEL,
+        color: RgbColor = DEFAULT_BACKGROUND_COLOR,
+    ) -> Optional[str]:
+        bg_label = find(self.items(), lambda x: x[1].color == color)
+        if bg_label is not None:
+            bg_label = bg_label[0]
+        else:
+            bg_label = name
+            if bg_label not in self:
+                bg_label = None
+
+        return bg_label
```

Comment:
I'd suggest to remove unnecessary reassignments for clarity:

```suggestion
        if bg_label is not None:
            return bg_label[0]

        if name in self:
            return name

        return None
```